src/content/app/genome-browser/state/browser-entity/browserEntitySlice.ts

```diff
@@ -0,0 +1,219 @@
+  const state = getState();
+  const activeEnsObjectIdsMap = getBrowserActiveEnsObjectIds(state);
+  Object.values(activeEnsObjectIdsMap).forEach((objectId) =>
+    dispatch(fetchEnsObject(objectId))
+  );
+};
+
+export const updateTrackStatesAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (payload: BrowserTrackStates) => (dispatch, getState: () => RootState) => {
+  dispatch(updateTrackStates(payload));
+  const trackStates = getBrowserTrackStates(getState());
+  browserStorageService.saveTrackStates(trackStates);
+};
+
+export const setDataFromUrlAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (payload: ParsedUrlPayload) => (dispatch) => {
+  dispatch(setEntityDataFromUrl(payload));
+  dispatch(setLocationDataFromUrl(payload));
+  dispatch(setTrackPanelDataFromUrl(payload));
+
+  const { activeGenomeId, activeEnsObjectId, chrLocation } = payload;
+
+  dispatch(ensureSpeciesIsEnabled(payload.activeGenomeId));
+  browserStorageService.saveActiveGenomeId(payload.activeGenomeId);
+  chrLocation &&
+    browserStorageService.updateChrLocation({ [activeGenomeId]: chrLocation });
+
+  if (activeEnsObjectId) {
+    browserStorageService.updateActiveEnsObjectIds({
+      [activeGenomeId]: activeEnsObjectId
+    });
+  }
+};
+
+export const updateBrowserActiveEnsObjectIdsAndSave = (
+  activeEnsObjectId: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    if (!activeGenomeId) {
+      return;
+    }
+    const currentActiveEnsObjectIds = getBrowserActiveEnsObjectIds(state);
+    const updatedActiveEnsObjectIds = {
+      ...currentActiveEnsObjectIds,
+      [activeGenomeId]: activeEnsObjectId
+    };
+
+    dispatch(updateBrowserActiveEnsObjectIds(updatedActiveEnsObjectIds));
+    dispatch(fetchEnsObject(activeEnsObjectId));
+
+    browserStorageService.updateActiveEnsObjectIds(updatedActiveEnsObjectIds);
+  };
+};
+
+export const deleteSpeciesInGenomeBrowser = (
+  genomeIdToRemove: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+
+    dispatch(deleteGenomeEntityData(genomeIdToRemove));
+    dispatch(deleteGenomeLocationData(genomeIdToRemove));
+    dispatch(deleteGenomeTrackPanelData(genomeIdToRemove));
+
+    const updatedActiveEnsObjectIds = pickBy(
+      getBrowserActiveEnsObjectIds(state),
+      (value, key) => key !== genomeIdToRemove
+    );
+
+    dispatch(updateBrowserActiveEnsObjectIds(updatedActiveEnsObjectIds));
+
+    browserStorageService.deleteGenome(genomeIdToRemove);
+    trackPanelStorageService.deleteGenome(genomeIdToRemove);
+  };
+};
+
+const browserEntitySlice = createSlice({
+  name: 'genome-browser-entity',
+  initialState: defaultBrowserEntityState as BrowserEntityState,
```

Comment:
`as BrowserEntityState` shouldn't be necessary; you already defined the type of `defaultBrowserEntityState` above.

Reply:
That's done